charmstore.go

```diff
@@ -239,6 +248,148 @@ func (s *CharmStore) Latest(curls ...*charm.URL) ([]CharmRevision, error) {
 	return responses, nil
 }
 
+// ListResources returns metadata for all the resources defined on the given charms.
+func (s *CharmStore) ListResources(ids []*charm.URL) ([]ResourceResult, error) {
+	if len(ids) == 0 {
+		return nil, nil
+	}
+
+	results, err := s.client.ListResources(ids)
+	if err != nil {
+		return nil, errgo.NoteMask(err, "cannot get resource metadata from the charm store", errgo.Any)
+	}
+
+	result := make([]ResourceResult, len(ids))
+	for i, id := range ids {
+		resources, ok := results[id.String()]
+		if !ok {
+			result[i].Err = CharmNotFound(id.String())
+			continue
+		}
+		list := make([]resource.Resource, len(resources))
+		for j, res := range resources {
+			resource, err := apiResource2Resource(res)
```

Comment:
Different API... that's the controller API.  In theory, we could make them look mostly the same, but I think there's always going to be differences, so I'm not sure it's worth trying to shoehorn them into the same mold.